File: php/features.php
<?php
declare(strict_types=1);

// OpenMeteo SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OpenMeteoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OpenMeteoBaseFeature();
            case "test":
                return new OpenMeteoTestFeature();
            default:
                return new OpenMeteoBaseFeature();
        }
    }
}

File: py/openmeteo_sdk/features.py
# OpenMeteo SDK feature factory

from openmeteo_sdk.feature.base_feature import OpenMeteoBaseFeature
from openmeteo_sdk.feature.test_feature import OpenMeteoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenMeteoBaseFeature(),
        "test": lambda: OpenMeteoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
